tiledb/sm/rest/rest_profile.cc

```diff
@@ -0,0 +1,286 @@
+/**
+ * @file   rest_profile.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2025 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file implements class RestProfile.
+ */
+
+#include <iostream>
+
+#include "rest_profile.h"
+#include "tiledb/common/random/random_label.h"
+#include "tiledb/sm/misc/constants.h"
+
+using namespace tiledb::common;
+using namespace tiledb::common::filesystem;
+
+namespace tiledb::sm {
+
+/* ****************************** */
+/*           STATIC API           */
+/* ****************************** */
+
+/**
+ * Read the given file and return its contents as a json object.
+ *
+ * @param filepath The path of the file to load.
+ * @return The contents of the file, as a json object.
+ */
+static json read_file(const std::string& filepath) {
+  json data;
+  {
+    std::ifstream file(filepath);
+    try {
+      file >> data;
+    } catch (...) {
+      throw RestProfileException("Error parsing file \'" + filepath + "\'.");
+    }
+  }
+  return data;
+}
+
+/**
+ * Write the given json to the given file.
+ *
+ * @param data The json data to write to the file.
+ * @param filepath The path of the file to which the data is written.
+ */
+static void write_file(json data, const std::string& filepath) {
+  // Temporarily append filepath with a random label to guarantee atomicity.
+  std::string temp_filepath = filepath + "." + random_label();
+
+  // Load the json object contents into the file.
+  std::ofstream file(temp_filepath);
+  try {
+    file << std::setw(2) << data;
+    file.flush();
+    file.close();
+  } catch (...) {
+    throw RestProfileException(
+        "Failed to write file due to an internal error during write.");
+  }
+
+  // Remove the random label from the filepath.
+  if (std::rename(temp_filepath.c_str(), filepath.c_str()) != 0) {
```

Comment:
I think we should we use `std::filesystem::rename` instead of this C-style rename.
Then we should catch exceptions and print the corresponding errors (`.what()`) , and we can debug issues like the one failing CI right now easier.